tutorials/getting_started/configuration.md

```diff
@@ -0,0 +1,344 @@
+
+```
+word1{wtd}tag1{td}word2{wtd}tag2{td}...{td}wordn{wtd}tagn
+```
+
+For some "word tag delimiter" `{wtd}` and some "token delimiter" `{td}`.
+
+_Our_ data files look like
+
+```
+The/at detectives/nns placed/vbd Barco/np under/in arrest/nn
+```
+
+which is why we need to specify
+
+```js
+    "word_tag_delimiter": "/",
+```
+
+We don't need to specify anything for the "token delimiter",
+since the default split-on-whitespace behavior is already correct.
+
+If you look at the code for `SequenceTaggingDatasetReader.read()`,
+it turns each sentence into a `TextField`
+of tokens and a `SequenceLabelField` of tags. The latter isn't
+really configurable, but the former wants a dictionary of
+[TokenIndexer](http://docs.allennlp.org/en/latest/api/allennlp.data.token_indexers.html#allennlp.data.token_indexers.token_indexer.TokenIndexer)s
+that indicate how to convert the tokens into arrays.
+
+Our configuration specifies two token indexers:
+
+```js
+    "token_indexers": {
+      "tokens": {
+        "type": "single_id",
+        "lowercase_tokens": true
+      },
+      "token_characters": {
+        "type": "characters"
+      }
+    }
+```
+
+The first, `"tokens"`, is a
+[`SingleIdTokenIndexer`](http://docs.allennlp.org/en/latest/api/allennlp.data.token_indexers.html#allennlp.data.token_indexers.single_id_token_indexer.SingleIdTokenIndexer)
+that just represents each token (word) as a single integer.
+The configuration also specifies that we *lowercase* the tokens before encoding;
+that is, that this token indexer should ignore case.
+
+The second, `"token_characters"`, is a
+[`TokenCharactersIndexer`](http://docs.allennlp.org/en/latest/api/allennlp.data.token_indexers.html#allennlp.data.token_indexers.token_characters_indexer.TokenCharactersIndexer)
+that represents each token as a list of int-encoded characters.
+
+Notice that this gives us two different encodings for each token.
+Each encoding has a "namespace", in this case `"tokens"` and `"token_characters"`.
+The `SequenceLabelField` also has a namespace, `"labels"`.
+
+## non-padded namespaces
+
+Most namespaces are "padded", which means that in addition to whatever tokens
+are represented in the training data they also contain `"@@PADDING@@"` and
+`"@@UNKNOWN@@"` tokens.  You may not want this behavior for some namespaces;
+in particular, a "labels" namespace will probably never any kind of sequence padding,
+and is likely to have every possible token represented in the training set.
+
+To this end, you can specify "non padded namespaces".
+The default non-padded namespaces are `("*tags", "*labels")`;
+that is, any namespace that ends with `tags` or `labels`.
+
+It turns out that this tutorial dataset does have a small number of POS tags
+in the validation data that don't show up in the training data. This means
+that we need to override the list of non-padded namespaces:
+
+```js
+  "non_padded_namespaces": [],
```

Comment:
Are we doing this to get padding, or simply `@@UNKNOWN@@` tokens for certain postags?